adapters/Unity/UnityAdapter/GADMediationAdapterUnity.m

```diff
@@ -14,83 +14,125 @@
 
 #import "GADMediationAdapterUnity.h"
 #import <UnityAds/UnityAds.h>
-#import "GADMAdapterUnity.h"
-#import "GADMAdapterUnityConstants.h"
-#import "GADMAdapterUnityRewardedAd.h"
 #import "GADMAdapterUnityUtils.h"
+#import "GADUnityRouter.h"
+#import "GADMUnityInterstitialMediationAdapterProxy.h"
+#import "GADMUnityRewardedMediationAdapterProxy.h"
+#import "GADMUnityBannerMediationAdapterProxy.h"
+#import "GADMediationConfiguration+Settings.h"
+#import "NSError+Unity.h"
 
-@interface GADMediationAdapterUnity ()
-
-@property(nonatomic, strong) GADMAdapterUnityRewardedAd *rewardedAd;
-
+@interface GADMediationAdapterUnity () <GADMediationRewardedAd, GADMediationInterstitialAd, GADMediationBannerAd>
+@property (nonatomic, strong) NSString *placementId;
+@property (nonatomic, strong) GADUnityBaseMediationAdapterProxy *adapterProxy;
+@property (nonatomic, strong) UADSBannerView *bannerView;
 @end
 
 @implementation GADMediationAdapterUnity
 
 // Called on Admob->init
-+ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
-             completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  NSMutableSet *gameIDs = [[NSMutableSet alloc] init];
-  for (GADMediationCredentials *cred in configuration.credentials) {
-    NSString *gameIDFromSettings = cred.settings[kGADMAdapterUnityGameID];
-    GADMAdapterUnityMutableSetAddObject(gameIDs, gameIDFromSettings);
-  }
-
-  if (!gameIDs.count) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorInvalidServerParameters,
-        @"UnityAds mediation configurations did not contain a valid game ID.");
-    completionHandler(error);
-    return;
-  }
-
-  NSString *gameID = [gameIDs anyObject];
-  if (gameIDs.count > 1) {
-    NSLog(@"Found the following game IDs: %@. "
-          @"Please remove any game IDs you are not using from the AdMob UI.",
-          gameIDs);
-    NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
-  }
-
-  [[GADMAdapterUnity alloc] initializeWithGameID:gameID withCompletionHandler:completionHandler];
++ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
+    NSSet *gameIDs = configuration.gameIds;
+    if (!gameIDs.count) {
+        completionHandler([NSError noValidGameId]);
+        return;
+    }
+    
+    NSString *gameID = [gameIDs anyObject];
+    if (gameIDs.count > 1) {
+        NSLog(@"Found the following game IDs: %@. "
+              @"Please remove any game IDs you are not using from the AdMob UI.",
+              gameIDs);
+        NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
```

Comment:
Interesting log. I guess this is because we dont allow multiple game IDs. right.
Should we use NSLog or maybe use the system provided by Admob for logs?

Reply:
Couldn't find anything in documentation regarding admob logs. Checked other adapters, everyone is using NSLog. So I guess it's fine?

Reply:
Maybe I mixed it with a different adapter.
Well if this is their approach then I guess ok